src/main/java/org/fenixedu/academictreasury/util/ExcelUtils.java

```diff
@@ -41,118 +41,88 @@
-            
-            if (sheet == null) {
-                throw new RuntimeException("error.ExcelUploadComponent.invalid.spreadsheet");
-            }
-            
-            final List<List<String>> spreadsheetContent = new ArrayList<List<String>>();
-            
-            XSSFRow row;
-            for(int i = sheet.getFirstRowNum(); i <= sheet.getLastRowNum(); i++) {
-                row = sheet.getRow(i);
-                
+        public TreasuryDefaultExcelSheetProcessor(int maxCols) {
+            super();
+            setRowProcessor(row -> {
                 final ArrayList<String> rowContent = new ArrayList<String>();
                 spreadsheetContent.add(rowContent);
-                
+
                 if(row == null) {
                     for(int j = 0; j < maxCols; j++) {
                         rowContent.add("");
                     }
-                    
-                    continue;
+
+                    return;
                 }
-                
-                XSSFCell cell;
+
+                Cell cell;
                 for(int j = 0; j < maxCols; j++) {
                     cell = row.getCell(j);
-                    
+
                     if(cell == null) {
                         rowContent.add("");
                         continue;
                     }
-                    
+
                     if (XSSFCell.CELL_TYPE_NUMERIC == cell.getCellType() && DateUtil.isCellDateFormatted(cell)) {
                         rowContent.add(new SimpleDateFormat("dd/MM/yyyy HH:mm").format(cell.getDateCellValue()));
                     } else {
                         cell.setCellType(XSSFCell.CELL_TYPE_STRING);
                         String value = cell.getStringCellValue();
                         rowContent.add(value);
                     }
-                    
+
                 }
-            }
-            
-            result.add(new ExcelSheet(sheet.getSheetName(), spreadsheetContent));
+            });
+        }
+
+        @Override
+        protected void beforeSheetProcess(Sheet sheet) {
+            super.beforeSheetProcess(sheet);
+            this.spreadsheetContent =  new ArrayList<List<String>>();
+        }
+
+        public List<List<String>> getSpreadsheetContent() {
+            return spreadsheetContent;
+        }
+
+        @Override
+        protected void afterSheetProcess(Sheet sheet) {
+            super.afterSheetProcess(sheet);
+            this.result.add(new ExcelSheet(sheet.getSheetName(), getSpreadsheetContent()));
         }
-        
-        return result;
+
+        public List<ExcelSheet> getResult() {
+            return this.result;
+        }
+    }
+	
+    public static List<List<String>> readExcel(final InputStream stream, int maxCols) throws IOException {
+        TreasuryDefaultExcelSheetProcessor sheetProcessor = new TreasuryDefaultExcelSheetProcessor(maxCols);
+        ExcelUtil.importExcel(XlsType.XLSX, stream, sheetProcessor, maxCols);
```

Comment:
@anilmamede Should we support xls files too, or xlsx is enough?

Reply:
The xls is very old. XLSX should be enough